### PR TITLE
change resource check to rb file method

### PR DESCRIPTION
**Pull request by lihongyan1**
Depend on PR:
https://github.com/openshift/cucushift/pull/8043

### CODE REVIEW

**Comment by lihongyan1** [comment]
Solved all the comments and trigger v3 job:
https://mastern-jenkins-csb-openshift-qe.cloud.paas.psi.redhat.com/job/Runner-v3/134657/console
@akostadinov @juzhao Please help review

File: lib/openshift/prometheus.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+require 'openshift/project_resource'
+
+module BushSlicer
+  class Prometheus < ProjectResource
+    RESOURCE = "prometheus"
+
+    def enforced_sample_limit(user: nil, cached: true, quiet: false)
+      raw_resource(user: user, cached: cached, quiet: quiet).
+        dig("spec", "enforcedSampleLimit")
```
Comment (pruan-rht) [inline lib/openshift/prometheus.rb:9]: @lihongyan1 should be on the same line as line 8

File: tools/find_image.rb [hunk 2]
```diff
@@ -30,7 +30,7 @@ def run
       c.action do |args, options|
         options.default \
           :build_type => "nightly-",
-          :url => "https://openshift-release.svc.ci.openshift.org/"
+          :url => "https://openshift-release.apps.ci.l2s4.p1.openshiftapps.com/"
```
Comment (pruan-rht) [inline tools/find_image.rb:33]: @lihongyan1 this file change doesn't seem to belong here but in a separate PR.  Also, please don't change the default url...the script should follow the url direct.

File: lib/openshift/prometheus.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require 'openshift/project_resource'
+
+module BushSlicer
+  class Prometheus < ProjectResource
+    RESOURCE = "prometheus"
```
Comment (akostadinov) [inline lib/openshift/prometheus.rb:5]: could you set the full qualified API here? That is with `.openshift.io` or whatever it is for this resource.

File: lib/openshift/thanos_ruler.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+require 'openshift/project_resource'
+
+module BushSlicer
+  class ThanosRuler < ProjectResource
+    RESOURCE = "thanosruler"
```
Comment (akostadinov) [inline lib/openshift/thanos_ruler.rb:5]: full qualified API name here too

File: lib/openshift/validating_webhook_configuration.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+require 'openshift/cluster_resource'
+
+module BushSlicer
+  class ValidatingWebhookConfiguration < ClusterResource
+    RESOURCE = "ValidatingWebhookConfiguration"
```
Comment (akostadinov) [inline lib/openshift/validating_webhook_configuration.rb:5]: full qualified API name here too,
please post `oc get -o yaml` output of resource in case you find trouble figuring that out.

File: lib/openshift/prometheus.rb (hunk 3 above)
Comment (lihongyan1) [reply lib/openshift/prometheus.rb:5]: `oc get prometheus -n openshift-monitoring -oyaml
apiVersion: v1
items:
- apiVersion: monitoring.coreos.com/v1
  kind: Prometheus
  metadata:
    creationTimestamp: "2020-08-19T06:29:20Z"
    generation: 1
    labels:
      prometheus: k8s
    managedFields:
    - apiVersion: monitoring.coreos.com/v1
`
I change the line as
`
  RESOURCE = "prometheus,monitoring.coreos.com"
`
Get the following error:
`waiting for operation up to 3600 seconds..
    And I wait for the "user-workload" prometheus to appear up to 120 seconds                                                                       # features/step_definitions/resources.rb:144
      hash not from a Prometheus: expected  but was Prometheus (RuntimeError)
      /home/hongyli/openshift/projects/verification-tests/lib/openshift/resource.rb:140:in `update_from_api_object'
      /home/hongyli/openshift/projects/verification-tests/lib/openshift/resource.rb:82:in `get'
      /home/hongyli/openshift/projects/verification-tests/lib/openshift/resource.rb:43:in `exists?'
      /home/hongyli/openshift/projects/verification-tests/lib/openshift/resource.rb:201:in `block in wait_to_appear'
      /home/hongyli/openshift/projects/verification-tests/lib/base_helper.rb:153:in `wait_for'
      /home/hongyli/openshift/projects/verification-tests/lib/openshift/resource.rb:200:in `wait_to_appear'
      /home/hongyli/openshift/projects/verification-tests/features/step_definitions/resources.rb:149:in `/^(I|admin) waits? for the "(.+?)" (\w+) to appear(?: in the(?: "(.+?)")? project)?(?: up to (\d+) seconds)?$/'
      ./features/tierN/monitoring/ci.feature:1195:in `And I wait for the "user-workload" prometheus to appear up to 120 seconds'
    And I wait for the "user-workload" thanos_ruler to appear up to 120 seconds                                                                     # features/step_definitions/resources.rb:144
      #set log level`

File: lib/openshift/thanos_ruler.rb (hunk 4 above)
Comment (lihongyan1) [reply lib/openshift/thanos_ruler.rb:5]: similar error as above

File: lib/openshift/validating_webhook_configuration.rb (hunk 5 above)
Comment (lihongyan1) [reply lib/openshift/validating_webhook_configuration.rb:5]: Similar error for full qualified API
For post `oc get -o yaml` I changed code, not sure correct, please check

File: lib/openshift/prometheus.rb (hunk 3 above)
Comment (akostadinov) [reply lib/openshift/prometheus.rb:5]: Could you paste full YAML output? The `prometheus` part may need plural but it is not apparent from current output.

wrt the error, are you using latest repo code? I think such error was fixed some months ago. See lib/openshift/etcd_cluster.rb for example.

Comment (akostadinov) [reply lib/openshift/prometheus.rb:5]: Actually looked at the code and the error perhaps comes exactly because "prometheus.monitoring.coreos.com" is incorrect. Please paste full YAML and I will be able to tell (or will be able to reproduce).

Comment (lihongyan1) [reply lib/openshift/prometheus.rb:5]: Full YAML
http://pastebin.test.redhat.com/897376

File: lib/openshift/validating_webhook_configuration.rb (hunk 5 above)
Comment (lihongyan1) [reply lib/openshift/validating_webhook_configuration.rb:5]: Full YAML file
http://pastebin.test.redhat.com/897377

File: lib/openshift/thanos_ruler.rb (hunk 4 above)
Comment (lihongyan1) [reply lib/openshift/thanos_ruler.rb:5]: Full yaml
http://pastebin.test.redhat.com/897380

File: lib/openshift/prometheus.rb (hunk 3 above)
Comment (akostadinov) [reply lib/openshift/prometheus.rb:5]: Cool, so look at `selfLink`, it is `prometheuses`. Thus `RESOURCE = "prometheuses.monitoring.coreos.com"`. You can check `selfLink` for the other resources also.

Comment (akostadinov) [reply lib/openshift/prometheus.rb:5]: Fixed an issue with f960862 so please rebase before you continue.
